docs/lifecycle-management.md

```diff
@@ -0,0 +1,185 @@
+# Higher level VM Management
+
+## Motivation
+
+So far KubeVirt provides a pod-like interface when managing VMs. This seems to
+be a good building block.
+On the other hand this might not be sufficient for usual virtualization
+management UIs, which do expect to manage the complete life-cycle of a VM,
+including shut down VMs 8which is currently not intended by the core runtime).
+
+
+This document outlines how minimal assumptions can be met and also visits how
+other requirements impact the design.
+
+
+## Requirements
+* Running and non-running VMs are handled through the same API
+* Creating a VM must be separate from launching a VM
+* Support for “flavor” based VMs (templating)
+* Support for expanding sparse VM definitions
+* Support for detailed VM definitions
+* Support hot-plugging
+* Representation of dynamic instance specific data
+* Support for defaulting (to support OS compatibility)
+Considerations
+* Sub-resources for actions
+* Presets to add certain configurations
+
+
+## Definitions
+* **Defaulting** - Setting certain defaults during expansion (i.e. virtio for a
+  bus type, or e1000 for as a nic model)
+* **Expansion** - VM definitions can be provided in a sparse way i.e. just a
+  disk. Once the VM is getting started, the runtime will infer additional
+  requirements for this device and add them to the runtime VM definition.
+  This process of inferring and adding additional devices is called expansion.
+* **Flavor** - Comparable to a template or a specific instance type
```

Comment:
I think it needs much more information around it. Flavours in the openstack world do a lot of different jobs. They allow the cloud admin to constrain tenants to running VMs with a certain "size". In theory this allows them to pick sizes that pack nicely into their available compute node resources. It is also done so that billing rules & charging levels can be kept quite simple. The coarse sizes can also force users of a particular feature to a higher cost tier than they would otherwise choose. Flavours thus coarsely control what hardware is enabled for a virtual machine. There can be alot of metadata added to flavours to allow fine grained hardware control by administrators. This is used to both fine tune the VM hardware config, but also as an additional level of access control. Many hardware tunables can be  set on both the flavour and image, but if set on the flavour by the admin, the tenant is prevented from overridding them on the image.